File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/IdentityKeyArgs.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.entity.key.LocalKey;
import com.bloock.sdk.entity.key.ManagedKey;

/**
 * Represents arguments for configuring an issuer key.
 */
public class IdentityKeyArgs {

  LocalKey localKey;
  ManagedKey managedKey;

  /**
   * Constructs an IdentityKeyArgs object with a local key.
   * @param localKey
   */
  public IdentityKeyArgs(LocalKey localKey) {
    this.localKey = localKey;
  }

  /**
   * Constructs an IdentityKeyArgs object with a managed key.
   * @param managedKey
   */
  public IdentityKeyArgs(ManagedKey managedKey) {
    this.managedKey = managedKey;
  }
}

File: languages/php/Bloock/Entity/IdentityV2/IdentityKeyArgs.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\Entity\Key\LocalKey;
use Bloock\Entity\Key\ManagedKey;
use Exception;

/**
 * Represents arguments for configuring an issuer key.
 */
class IdentityKeyArgs
{
    public ?LocalKey $localKey = null;
    public ?ManagedKey $managedKey = null;

    /**
     * Constructs an IdentityKeyArgs object with a local or managed key.
     * @param $key
     * @throws Exception
     */
    public function __construct($key)
    {
        if ($key instanceof LocalKey) {
            $this->localKey = $key;
        } else if ($key instanceof ManagedKey) {
            $this->managedKey = $key;
        } else {
            throw new Exception("Invalid $key provided. Must be of type LocalKey or ManagedKey");
        }
    }
}
